lib/board.rb

```diff
@@ -0,0 +1,86 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells=[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells=[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} \n\n"
+  end
+
+  def position(spot)
+    cells[spot.to_i-1]
+  end
+
+  def update(spot, player)
+      cells[spot.to_i-1] = player.token
+  end
+
+  def all_X
+    ex = []
+    cells.each_with_index do |p, i| 
+      if p == "X"
+        ex << i
+      end
+    end
+    ex
```

Comment:
Typically, the pattern of
```
instantiate new array (ex)
iterate through another array (cells) and update the new array (ex)
return new array (ex)
```
indicates that a higher level iterator can be used. This one's a little tricky, but I would recommend [`collect`](http://ruby-doc.org/core-2.3.0/Enumerable.html#method-i-collect) to first transform the array to contain the indices of the elements equaling 'X'.
```
cells.collect.with_index {|p,i| i if p == "X"}
```
Any elements that don't equal "X" will now be `nil`, something like `["X", "O", "X", " ", "X"," ","O",X","O"]` => `[0,nil,2,nil,4,nil,nil,7,nil]`. Then, to remove the nil elements, just use [`compact`](http://ruby-doc.org/core-2.3.0/Array.html#method-i-compact).
```
def all_x
  cells.collect.with_index {|p,i| i if p == "X"}.compact
end
```